### PR TITLE
Add link preview in the popover and fix size issue

### PR DESCRIPTION
**Pull request by Jinjiu96**

### CODE REVIEW

File: addons/html_editor/static/src/main/link/link_popover.js [hunk 1]
```diff
@@ -75,4 +91,95 @@ export class LinkPopover extends Component {
...
+            // would need to fetch the page through the server (s2s), involving
+            // enduser fetching problematic pages such as illicit content.
+            this.previewFaviconImg.el.src = `https://www.google.com/s2/favicons?sz=16&domain=${encodeURIComponent(
+                url
+            )}`;
+            this.previewFaviconImg.el.classList.remove("d-none");
+            this.previewFaviconFa.el.classList.add("d-none");
+        } else {
+            await this.keepLastPromise
+                .add($.get(this.state.href))
```
Comment (fdardenne) [inline addons/html_editor/static/src/main/link/link_popover.js]: avoid JQUery, maybe use `fetch`?

File: addons/html_editor/static/src/main/link/link_popover.js [hunk 2]
```diff
@@ -75,4 +91,95 @@ export class LinkPopover extends Component {
...
+            });
+            return;
+        }
+        this.resetPreview(url);
+        const protocol = url.protocol;
+        if (!protocol.startsWith("http")) {
+            const faMap = { "mailto:": "fa-envelope-o", "tel:": "fa-phone" };
+            const icon = faMap[protocol];
+            if (icon) {
+                this.previewFaviconFa.el.classList.toggle("fa-globe");
```
Comment (fdardenne) [inline addons/html_editor/static/src/main/link/link_popover.js]: In the template should be cleaner

File: addons/html_editor/static/src/main/link/link_popover.js [hunk 3]
```diff
@@ -75,4 +91,95 @@ export class LinkPopover extends Component {
...
+            "fa-phone"
+        );
+        this.previewFaviconFa.el.classList.add("fa-globe");
+        this.previewFaviconImg.el.classList.add("d-none");
+
+        this.urlLink.el.textContent = this.state.url || _t("No URL specified");
+        this.urlLink.el.href = this.state.url || null;
+
+        this.fullUrl.el.textContent = this.state.url || _t("No URL specified");
+        this.fullUrl.el.href = this.state.url || null;
```
Comment (fdardenne) [inline addons/html_editor/static/src/main/link/link_popover.js]: You can probably move this logic directly in the template too !

File: addons/html_editor/static/src/main/link/link_popover.xml [hunk 4]
```diff
@@ -4,22 +4,36 @@
...
-                <a href="#" class="ms-1 o_we_remove_link text-dark" t-on-click="onClickRemove" data-bs-toggle="tooltip" data-bs-placement="top" title="${('Remove Link')}">
-                    <i class="fa fa-chain-broken"></i>
-                </a>
+                <span class="me-2 o_we_preview_favicon">
+                    <i t-attf-class="fa {{state.faIcon}} {{state.previewImg ? 'd-none' : ''}}"></i>
+                    <img t-attf-src="{{state.imgSrc}}" t-attf-class="align-baseline {{state.previewImg ? '' : 'd-none'}}"></img>
+                </span>
+                <div class="w-100">
+                    <div class="d-flex">
+                        <a href="#" target="_blank" t-attf-href="{{state.url || null}}" class="o_we_url_link fw-bold flex-grow-1 text-truncate" title="Open in a new tab">
```
Comment (fdardenne) [inline addons/html_editor/static/src/main/link/link_popover.xml]: Hmm I'm wondering if you can't just put 
```suggestion
                        <a href="#" target="_blank" t-attf-href="{{state.url}}" class="o_we_url_link fw-bold flex-grow-1 text-truncate" title="Open in a new tab">
```

**Review by fdardenne — COMMENTED** [review]
That's a lot cleaner good job ! 
One more thing, you can also replace `d-none` by a `t-if` 💯

File: addons/html_editor/static/src/main/link/link_popover.xml (hunk 4 above)
Comment (Jinjiu96) [reply addons/html_editor/static/src/main/link/link_popover.xml]: Ahh indeed, already assigned null when initializing state.url :rofl:

**Pull request merged**